lib/openstack_query/query_wrapper.py

```diff
@@ -0,0 +1,353 @@
+    prop_not_equal_to,
+    prop_greater_than,
+    prop_less_than,
+    prop_younger_than,
+    prop_older_than,
+    prop_older_than_or_equal_to,
+    prop_younger_than_or_equal_to,
+    prop_any_in,
+    prop_not_any_in,
+    prop_matches_regex,
+)
+
+
+from exceptions.parse_query_error import ParseQueryError
+from exceptions.query_mapping_error import QueryMappingError
+
+from enums.query.query_presets import QueryPresets
+from tabulate import tabulate
+
+
+class QueryWrapper(OpenstackWrapperBase):
+    _PROPERTY_MAPPINGS = {}
+    _KWARG_MAPPINGS = {}
+    _NON_DEFAULT_FILTER_FUNCTION_MAPPINGS = {}
+    _DEFAULT_FILTER_FUNCTION_MAPPINGS = {}
+    _DEFAULT_FILTER_FUNCTIONS = {
+        QueryPresets.EQUAL_TO: prop_equal_to,
+        QueryPresets.NOT_EQUAL_TO: prop_not_equal_to,
+        QueryPresets.GREATER_THAN: prop_greater_than,
+        QueryPresets.LESS_THAN: prop_less_than,
+        QueryPresets.YOUNGER_THAN: prop_younger_than,
+        QueryPresets.YOUNGER_THAN_OR_EQUAL_TO: prop_younger_than_or_equal_to,
+        QueryPresets.OLDER_THAN_OR_EQUAL_TO: prop_older_than_or_equal_to,
+        QueryPresets.OLDER_THAN: prop_older_than,
+        QueryPresets.ANY_IN: prop_any_in,
+        QueryPresets.NOT_ANY_IN: prop_not_any_in,
+        QueryPresets.MATCHES_REGEX: prop_matches_regex,
+    }
+
+    def __init__(self, connection_cls=OpenstackConnection):
+        OpenstackWrapperBase.__init__(self, connection_cls)
+        self._query_props = dict()
+        self._filter_func = None
+        self._filter_kwargs = None
+        self._results_resource_objects = []
+        self._results = []
+
+    def select(self, *props: Enum):
+        """
+        Public method used to 'select' properties that the query will return the value of.
+        Mutually exclusive to returning objects using select_all()
+        :param props: one or more properties to collect described as enum
+        """
+
+        # should be a idempotent function
+        # calling multiple times with should aggregate properties to select
+        for prop in props:
+            prop_name, prop_func = self._get_prop(prop)
+            self._query_props[prop_name] = prop_func
+        return self
+
+    def _get_prop(self, prop: Enum) -> Tuple[str, Callable[[Any], Any]]:
+        """
+        static method which returns a property name and 'get_property' function for a given property enum.
+        This 'get_property' function takes a Openstack resource object and returns a value which
+        corresponds to the property given of that object
+        """
+        if prop not in self._PROPERTY_MAPPINGS.keys():
+            raise QueryMappingError(
+                """
+                Error: failed to get property mapping, the property is valid
+                but does not contain an entry in PROPERTY_MAPPINGS.
+                Please raise an issue with repo maintainer
+            """
+            )
+        return prop.name.lower(), self._PROPERTY_MAPPINGS[prop]
```

Comment:
Nit: use `.casefold()` instead of lower when doing case insensitive comparisons on Python 3+